Packs/Aws-SecretsManager/Integrations/AwsSecretsManager/AwsSecretsManager.py

```diff
@@ -0,0 +1,285 @@
+    )
+
+    kwargs = dict()
+
+    if args.get('secret_id'):
+        kwargs['SecretId'] = args.get('secret_id')
+    if args.get('days_of_recovery'):
+        kwargs['RecoveryWindowInDays'] = int(args.get('days_of_recovery'))
+    if args.get('delete_immediately') is not None:
+        if args.get('days_of_recovery'):
+            return_error('Delete command cannot be executed with both args: delete_immediately and days_of_recovery')
+        kwargs['ForceDeleteWithoutRecovery'] = args.get('delete_immediately', False) == True
+
+    response = client.delete_secret(**kwargs)
+
+    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+        demisto.results("The Secret was Deleted")
+
+def aws_secrets_manager_secret_restore_command(client: AWSClient, args: Dict[str, Any]) -> CommandResults:
+    client = client.aws_session(
+        service=SERVICE,
+        role_arn=args.get('roleArn'),
+        role_session_name=args.get('roleSessionName'),
+        role_session_duration=args.get('roleSessionDuration'),
+    )
+
+    if not args.get('secret_id'):
+        return_error('secret_id is mandatory inorder to run this command!')
+
+    response = client.restore_secret(SecretId=args.get('secret_id'))
+
+    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+        demisto.results("the secret was restored successfully")
+
+def aws_secrets_manager_secret_policy_get_command(client: AWSClient, args: Dict[str, Any]) -> CommandResults:
+    client = client.aws_session(
+        service=SERVICE,
+        role_arn=args.get('roleArn'),
+        role_session_name=args.get('roleSessionName'),
+        role_session_duration=args.get('roleSessionDuration'),
+    )
+
+    if not args.get('secret_id'):
+        return_error('secret_id is mandatory inorder to run this command!')
+
+    response = client.get_resource_policy(
+        SecretId=args.get('secret_id')
+    )
+
+    readable_output = {'Name': response.get('Name', ''),
+                       'ARN': response.get('ARN', ''),
+                       'Policy': response.get('ResourcePolicy', '')}
+
+    human_readable = tableToMarkdown('AWS Secret Policy', readable_output)
+
+    return_results(CommandResults(
+        outputs_prefix='AWS.SecretsManager.Policy',
+        outputs=response,
+        readable_output=human_readable
+    ))
+
+def fetch_credentials(client: AWSClient, args: Dict[str, Any]) -> CommandResults:
+    client = client.aws_session(
+        service=SERVICE,
+        role_arn=args.get('roleArn'),
+        role_session_name=args.get('roleSessionName'),
+        role_session_duration=args.get('roleSessionDuration'),
+    )
+
+    creds_dict = dict()
+    if args.get('secret_id'):
+        try:
+            creds_dict[args.get('secret_id')] = client.get_secret_value(SecretId=args.get('secret_id'))
+        except Exception as e:
+            demisto.debug(f"Could not fetch credentials: {args.get('secret_id')}. Error: {e}")
```

Comment:
I believe in this case it's better to raise this as an error, as the user won't see this error.
@yuvalbenshalom wdyt?